packages/backend/src/open_payments/quote/service.ts

```diff
@@ -271,6 +271,11 @@ export async function finalizeQuote(
     'Content-Type': 'application/json'
   }
 
+  const receiver = await deps.clientService.receiver.get(quote.receiver)
```

Comment:
We shouldn't need to make another request to get the receiver here.
We could instead pass `receiver` as an argument to `finalizeQuote`.